projects/storefrontlib/src/cms-components/myaccount/update-email/update-email.component.ts

```diff
@@ -56,11 +56,14 @@ export class UpdateEmailComponent implements OnInit, OnDestroy {
       );
       // TODO(#9638): Use logout route when it will support passing redirect url
       await this.authService.coreLogout();
-      this.routingService.go({ cxRoute: 'login' }, null, {
-        state: {
-          newUid: this.newUid,
-        },
-      });
+      this.routingService.go(
+        { cxRoute: 'login' },
+        {
+          state: {
+            newUid: this.newUid,
+          },
+        }
+      );
```

Comment:
✅  Here using `state` is OK, becasue `newUid` it not a query param, but indeed the Angular `NavigationExtras.state`.